Source/bmalloc/libpas/src/libpas/pas_enumerator.c

```diff
@@ -462,4 +464,92 @@ bool pas_enumerator_enumerate_all(pas_enumerator* enumerator)
     return true;
 }
 
+static crash_reporter_memory_reader_t memory_reader;
+
+static kern_return_t
+memory_reader_adapter(task_t task, vm_address_t address, vm_size_t size, void **local_memory)
+{
+    void *ptr = memory_reader(task, address, size);
+    *local_memory = ptr;
+    return ptr ? KERN_SUCCESS : KERN_FAILURE;
+}
+
+static memory_reader_t *
+setup_memory_reader(crash_reporter_memory_reader_t crm_reader)
+{
+    memory_reader = crm_reader;
+    return memory_reader_adapter;
+}
+
+kern_return_t pas_pgm_extract_crash_report(vm_address_t fault_address, mach_vm_address_t pas_dead_root, addr64_t pgm_zone, task_t task, pas_pgm_report *report, crash_reporter_memory_reader_t crm_reader)
```

Comment:
This is might not compile on older OSes since this signature changed, so you'll have to provide a stub for older OSes with a different signature.

We actually could support PGM on older OSes since new WebKit ships to older macOS releases regularly—though we would also need to back-port other internal changes to support it.  So you could also just make this new code conditional on future OS releases.

Reply:
Which signature changed?

This is a new function and shouldn't have anything relying on it from the analytics side yet.

Reply:
Talked with David. This should not cause any signature changes 😀.